### ISSUE TITLE
Object detection training: GET was unable to find an engine to execute this computation

### ISSUE THREAD

**Issue by ERYAGNIK003** [ultralytics/ultralytics#3472]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I have followed below steps : 
1. Created virtual environment 
2. pip install ultralytics
3. Prepared data.yaml file for dataset

Then run given code for model training and got below errors : 

![Error1](https://github.com/ultralytics/ultralytics/assets/13064391/f9493d59-d2a6-4aad-9442-420318b9e5e7)
![error2](https://github.com/ultralytics/ultralytics/assets/13064391/f98048a8-df44-4c47-803d-3f1fd032260b)

### Environment

- YOLOv8 on Ubuntu 20.04.5 LTS (Quadro RTX 6000/8000)
- Python 3.11, Jupyter notebook, torch 2.0.1+cu117, torchvision 0.15.2+cu117
- NVIDIA drivers 
![nvidiasmi](https://github.com/ultralytics/ultralytics/assets/13064391/ac2a17cf-68d6-4cfe-926c-24d79fd1370d)

### Minimal Reproducible Example

from ultralytics import YOLO
# Load a model
model = YOLO("yolov8n.yaml")  # build a new model from scratch
# Train the model
model.train(data='data.yaml', epochs=100, imgsz=640,amp=False)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@ERYAGNIK003 I've never seen this error before. It's likely related to your environment. First I would try to train a default dataset like coco8.yaml, and if you get the same error then I'd start from one of the official environments below. Also see https://docs.ultralytics.com/quickstart for more detailed installation instructions.


## Environments

YOLOv8 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/quickstart) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

**Comment by ERYAGNIK003** [comment]
If i set amp=True in model than it is giving me same error... Are the NVIDIA driver versions and torch versions are ok ?

**Comment by glenn-jocher** [comment]
@ERYAGNIK003 it's possible that the issue is related to the versions of your NVIDIA drivers and Torch. To address this, you can try the following steps:

1. Ensure that you have the latest NVIDIA drivers installed on your system. You can check the official NVIDIA website for the most up-to-date drivers compatible with your GPU model.

2. Verify that you have the correct version of Torch installed. YOLOv8 is designed to work with PyTorch, so make sure you have the compatible version of PyTorch installed. You can refer to the PyTorch website or the YOLOv8 documentation for the recommended PyTorch version.

By updating your NVIDIA drivers and ensuring you have the correct version of Torch, you may be able to resolve the issue and successfully run YOLOv8 with the `amp=True` setting.

**Comment by ERYAGNIK003** [comment]
Sure. I will try to update NVIDIA drivers and PyTorch. Thanks.

**Comment by glenn-jocher** [comment]
@ERYAGNIK003 hello,

Based on the error you encountered, it is possible that the issue is related to the versions of your NVIDIA drivers and PyTorch. To address this, you can try the following steps:

1. Make sure you have the latest NVIDIA drivers installed on your system. Visit the official NVIDIA website to find the most up-to-date drivers that are compatible with your GPU model.

2. Verify that you have the correct version of PyTorch installed. YOLOv8 is designed to work with PyTorch, so it is important to have the compatible version. Consult the PyTorch documentation or the YOLOv8 documentation for the recommended PyTorch version.

By updating your NVIDIA drivers and ensuring you have the correct version of PyTorch installed, you may be able to resolve the issue and successfully train your YOLOv8 model.

I hope this helps! Let me know if you have any further questions.
